### PR TITLE
feat!: replace current text editor

### PR DESCRIPTION
**Pull request by harish-sethuraman**
## What does this PR do?

Replaces current text editor with new one

https://user-images.githubusercontent.com/32865581/199824651-308ae2ac-5cb0-4e8b-bcfe-95558cdbcd31.mov

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?
- test new changes features
- test existing texts as well

### CODE REVIEW

File: src/components/Composer/Editor/ToolbarPlugin.tsx [hunk 1]
```diff
@@ -51,19 +49,13 @@ export default function ToolbarPlugin() {
...
       if (elementDOM !== null) {
-        setSelectedElementKey(elementKey);
         if ($isListNode(element)) {
           const parentList = $getNearestNodeOfType<ListNode>(anchorNode, ListNode);
           const type = parentList ? parentList.getListType() : element.getListType();
           setBlockType(type);
         } else {
           const type = $isHeadingNode(element) ? element.getTag() : element.getType();
           setBlockType(type);
-          if ($isCodeNode(element)) {
```
Comment (harish-sethuraman) [inline src/components/Composer/Editor/ToolbarPlugin.tsx]: I left it thinking we might need this if we try to add code snippets

File: src/components/Composer/Editor/index.tsx [hunk 2]
```diff
@@ -108,25 +77,29 @@ export default function Editor() {
...
+          onChange={(editorState) => {
+            editorState.read(() => {
+              const markdown = $convertToMarkdownString(TRANSFORMERS);
+              setPublicationContent(markdown);
+            });
+          }}
         />
-        <OnChangePlugin onChange={onChange} />
         <HistoryPlugin />
-        <ListPlugin />
```
Comment (harish-sethuraman) [inline src/components/Composer/Editor/index.tsx]: This plugin allows you to nest the list and break out of list in markdown? Did you test it?

File: src/components/Composer/Editor/ToolbarPlugin.tsx (hunk 1 above)
Comment (bigint) [reply src/components/Composer/Editor/ToolbarPlugin.tsx]: Yup we can add incrementally 🙇🏼

**Comment by bigint** [comment]
Da resolve conflicts 🙏🏼

File: src/components/Composer/Editor/index.tsx (hunk 2 above)
Comment (bigint) [reply src/components/Composer/Editor/index.tsx]: Yes there is no limit of nesting :( sed

Comment (harish-sethuraman) [reply src/components/Composer/Editor/index.tsx]: fixed

**Comment by bigint** [comment]
We can remove this bg color 🙏🏼, it doesn't match with our brand 🙇🏼 

image

**Comment by harish-sethuraman** [comment]
Should we do a light purple?

**Comment by bigint** [comment]
> Should we do a light purple?

we can remove it

**Comment by harish-sethuraman** [comment]
This is to highlight that there is a mention in text editor. If we remove it it might kinda be a little confusing? 😅

**Comment by bigint** [comment]
@harish-sethuraman `publicationContent` not getting cleared when we call `setPublicationContent('')` 😕

**Review by bigint — APPROVED** [review]
Great work @harish-sethuraman 🖖🏼 

**LGTM** :shipit: 

Rad LGTM

**Pull request merged**